### PR TITLE
WFLY-10872 Confusing error message when writing simple-load-provider …

### PR DESCRIPTION
**Pull request by rhusar**
…attribute when dynamic load provider is present

Jira
https://issues.jboss.org/browse/WFLY-10872

### CODE REVIEW

File: mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ProxyConfigurationResourceDefinition.java [hunk 1]
```diff
@@ -428,6 +429,33 @@ public PathAddress convertToTargetAddress(PathAddress aliasAddress, AliasContext
             return SimpleLoadProviderResourceDefinition.Attribute.FACTOR;
         }
 
+        AttributeValueTranslator SIMPLE_LOAD_FACTOR_DEFINED_VALIDATOR = (context, value) -> {
```
Comment (pferraro) [inline mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ProxyConfigurationResourceDefinition.java]: Replace with anonymous class.

File: mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ProxyConfigurationResourceDefinition.java [hunk 2]
```diff
@@ -428,6 +429,33 @@ public PathAddress convertToTargetAddress(PathAddress aliasAddress, AliasContext
...
+            // Skip if executed on the existing current resource
+            PathAddress currentAddress = context.getCurrentAddress();
+            if (currentAddress.getLastElement().equals(SimpleLoadProviderResourceDefinition.PATH)) {
+                return value;
+            }
+
+            try {
+                // Otherwise, fail if operation executed on the legacy resource if dynamic load provider is defined
+                PathAddress address = PathAddress.pathAddress(DynamicLoadProviderResourceDefinition.PATH);
+                context.readResource(address);
```
Comment (pferraro) [inline mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ProxyConfigurationResourceDefinition.java]: Use non-recursive read, e.g. context.readResource(address, false)

**Review by pferraro — CHANGES_REQUESTED** [review]

File: mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ProxyConfigurationResourceDefinition.java (hunk 1 above)
Comment (rhusar) [reply mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ProxyConfigurationResourceDefinition.java]: Lambdas still no good? :)

File: mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ProxyConfigurationResourceDefinition.java (hunk 2 above)
Comment (rhusar) [reply mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ProxyConfigurationResourceDefinition.java]: Ah, thanks.

File: mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ProxyConfigurationResourceDefinition.java [hunk 3]
```diff
@@ -428,6 +429,36 @@ public PathAddress convertToTargetAddress(PathAddress aliasAddress, AliasContext
             return SimpleLoadProviderResourceDefinition.Attribute.FACTOR;
         }
 
+        AttributeValueTranslator SIMPLE_LOAD_FACTOR_DEFINED_VALIDATOR = new AttributeValueTranslator() {
```
Comment (pferraro) [inline mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ProxyConfigurationResourceDefinition.java]: This property should be final.  Also, use camel case for non-static properties.
SIMPLE_LOAD_PROVIDER_TRANSLATION, however, should probably be both static and final.

**Review by pferraro — CHANGES_REQUESTED** [review]

File: mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ProxyConfigurationResourceDefinition.java (hunk 3 above)
Comment (rhusar) [reply mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ProxyConfigurationResourceDefinition.java]: Ok, cannot be static here, so final will suffice.

Comment (pferraro) [reply mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ProxyConfigurationResourceDefinition.java]: Yes, but it is allowed for SIMPLE_LOAD_PROVIDER_TRANSLATION.

Comment (rhusar) [reply mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ProxyConfigurationResourceDefinition.java]: Got it now, yes it should, fixed.

**Comment by darranl** [comment]
@pferraro Are you Ok to double check this one?  GitHub is showing changes where you commented.

**Comment by pferraro** [comment]
@darranl yeah - I'll double check and approve when ready.

**Review by pferraro — APPROVED** [review]

**Pull request merged**